src/main/scala/com/yotpo/metorikku/output/writers/file/FileOutputWriter.scala

```diff
@@ -104,6 +107,13 @@ class FileOutputWriter(props: Map[String, Object], outputFile: Option[File]) ext
 
         writer.save()
 
+        fileOutputProperties.protectFromEmptyOutput match {
+          case Some(true) => {
+            if (catalog.tableExists(tableName) && dataFrame.sparkSession.table(tableName).head(1).isEmpty) {
+              throw MetorikkuWriteFailedException(s"Aborting Hive external table update -> ${tableName} is empty!")
```

Comment:
Wow this is something so small and so important!!

Reply:
I agree. This is life changing.